packages/fxa-payments-server/src/components/PaymentForm/index.test.tsx

```diff
@@ -197,6 +212,301 @@ it('includes the confirmation checkbox when confirm = true and plan supplied', (
+      const expectedMsg =
+        'I authorize Mozilla, maker of Firefox products, to charge my payment method <strong>$5.00 every 6 days</strong>, according to payment terms, until I cancel my subscription.';
+
+      const testRenderer = TestRenderer.create(
+        <Subject {...{ confirm: true, plan: plan }} />
+      );
+      const testInstance = testRenderer.root;
+      const legalCheckbox = testInstance.findByProps({ id: expectedMsgId });
+
+      expect(legalCheckbox.props.$amount).toBe('5.00');
+      expect(legalCheckbox.props.$intervalCount).toBe(6);
+      expect(legalCheckbox.props.children.props.children).toBe(expectedMsg);
+    });
+
+    it('renders Localized for weekly plan with correct props and displays correct default string', async () => {
+      const plan_id = 'plan_weekly';
+      const plan = findMockPlan(plan_id);
+      const expectedMsgId = 'payment-confirm-week';
+      const expectedMsg =
+        'I authorize Mozilla, maker of Firefox products, to charge my payment method <strong>$5.00 weekly</strong>, according to payment terms, until I cancel my subscription.';
+
+      const testRenderer = TestRenderer.create(
+        <Subject {...{ confirm: true, plan: plan }} />
+      );
+      const testInstance = testRenderer.root;
+      const legalCheckbox = testInstance.findByProps({ id: expectedMsgId });
+
+      expect(legalCheckbox.props.$amount).toBe('5.00');
+      expect(legalCheckbox.props.$intervalCount).toBe(1);
+      expect(legalCheckbox.props.children.props.children).toBe(expectedMsg);
+    });
+
+    it('renders Localized for 6 weeks plan with correct props and displays correct default string', async () => {
+      const plan_id = 'plan_6weeks';
+      const plan = findMockPlan(plan_id);
+      const expectedMsgId = 'payment-confirm-week';
+      const expectedMsg =
+        'I authorize Mozilla, maker of Firefox products, to charge my payment method <strong>$5.00 every 6 weeks</strong>, according to payment terms, until I cancel my subscription.';
+
+      const testRenderer = TestRenderer.create(
+        <Subject {...{ confirm: true, plan: plan }} />
+      );
+      const testInstance = testRenderer.root;
+      const legalCheckbox = testInstance.findByProps({ id: expectedMsgId });
+
+      expect(legalCheckbox.props.$amount).toBe('5.00');
+      expect(legalCheckbox.props.$intervalCount).toBe(6);
+      expect(legalCheckbox.props.children.props.children).toBe(expectedMsg);
+    });
+
+    it('renders Localized for monthly plan with correct props and displays correct default string', async () => {
+      const plan_id = 'plan_monthly';
+      const plan = findMockPlan(plan_id);
+      const expectedMsgId = 'payment-confirm-month';
+      const expectedMsg =
+        'I authorize Mozilla, maker of Firefox products, to charge my payment method <strong>$5.00 monthly</strong>, according to payment terms, until I cancel my subscription.';
+
+      const testRenderer = TestRenderer.create(
```

Comment:
The assert below appear to be common for all these tests. You could probably pull them out in a helper function.

Reply:
Yes, I agree.